crates/biome_service/src/workspace/scanner.rs

```diff
@@ -211,6 +213,35 @@ pub(crate) struct ScanContext<'app> {
 
     /// List of paths that should be processed.
     pub(crate) evaluated_paths: RwLock<BTreeSet<BiomePath>>,
+
+    /// What the scanner should target.
+    target: ScanKind,
```

Comment:
"Target" feels a bit too generic and potentially confusing here.

```suggestion
    /// The kind of scan to perform.
    scan_kind: ScanKind,
```